samza-sql/src/main/java/org/apache/samza/sql/SamzaSqlInputTransformer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.sql;
+
+import java.time.Instant;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.samza.operators.KV;
+import org.apache.samza.sql.data.SamzaSqlRelMsgMetadata;
+import org.apache.samza.system.IncomingMessageEnvelope;
+import org.apache.samza.system.descriptors.InputTransformer;
+
+
+/**
+ * SamzaSqlInputTransformer:
+ *   Given a userTransformer, this transformer applies the usreTransformer.apply
+ *   and when WrapMessage is true, it wraps transformed message into a SamzaMessageEnvelope
+ *   copying eventTime and arrivalTime from the input InputMessageEnvelope.
+ */
+
+public class SamzaSqlInputTransformer implements InputTransformer {
+
+  private boolean isKeyed; // output is KV or not, in case userTransformer is null
```

Comment:
This would also always be false, right? Since there is no userTransformer?

Reply:
I changed  to depend on the message itself, if it is not KV already, create one